code/processes/filealerter.q

```diff
@@ -93,19 +93,35 @@ chktable:{[files] table:([]filename:files;md5hash:$[usemd5;gethash'[files];(coun
 
 //-The main function that brings everything together
 processfiles:{[DICT]
-        /-find all matches to the file search string
-        matches:find[.rmvr.removeenvvar[DICT[`path]];DICT[`match]];
-        toprocess:getunprocessed[matches;DICT[`newonly]];
-        files:exec filename from toprocess;
-        /-If there are files to process
-        $[0<count files;
-        [{.lg.o[`alerter;"found file ", x]}'[files];
-        /-perform the function on the file
-        pf:action/:[DICT[`function];files];];
-        .lg.o[`alerter;"no new files found"]];
-        t:update function:(count toprocess)#DICT[`function],funcpassed:pf,moveto:(count toprocess)#enlist .rmvr.removeenvvar[DICT[`movetodirectory]] from toprocess; t}
-
-
+    /-find all matches to the file search string
+    matches:find[.rmvr.removeenvvar[DICT[`path]];DICT[`match]];
+    /-process duplicate files, either remove, move or leave depending on duplicateaction variable
+    processdups[;DICT] each matches;
+    toprocess:getunprocessed[matches;DICT[`newonly]];
```

Comment:
I think I would do this the other way around.  So the duplicates are basically

matches except toprocess

Otherwise we will end up with duplicate detection logic in two places.